orchagent/watermarkorch.cpp

```diff
@@ -170,16 +201,20 @@ void WatermarkOrch::doTask(SelectableTimer &timer)
 
     if (&timer == m_telemetryTimer)
     {
-        /* If the interval was changed */
-        auto intervT = timespec { .tv_sec = m_telemetryInterval , .tv_nsec = 0 };
-        m_telemetryTimer->setInterval(intervT);
-        m_telemetryTimer->reset();
+        /* Timer is only running when the watermark polling is on                         *
+         * if it is disabled we will not restart the timer at the end of current interval */
+        if (m_isTimerEnabled)
+        {
+            auto intervT = timespec { .tv_sec = m_telemetryInterval , .tv_nsec = 0 };
+            m_telemetryTimer->setInterval(intervT);
```

Comment:
We do not need to set the firing interval each time. Only when there is a change from CONFIG_DB.